dask_geopandas/geohash.py

```diff
@@ -0,0 +1,248 @@
+"""
+Geohash implementation
+
+The code is originally based on the neathgeohash package,
+Copyright (c) 2020 Marek Dwulit, MIT License
+(https://pypi.org/project/neathgeohash/#description).
+The vectorized implementation for quantization and bit interleaving is in turn based on,
+"Geohash in Golang Assembly" blog (https://mmcloughlin.com/posts/geohash-assembly).
+
+"""
+import numpy as np
+import pandas as pd
+from ._utils import _calculate_mid_points
+
+
+# Implementation based on deprecated https://pypi.org/project/neathgeohash/#description
+
+
+def _geohash(gdf, string, p):
+    """
+    Calculate geohash based on the middle points of the geometry bounds
+    for a given precision
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+    string : bool
+        to return string or int Geohash
+    p : int
+        precision of the string Geohash
+
+    Returns
+    ---------
+    type : pandas.Series
+        Series containing geohash
+    """
+
+    # Calculate bounds
+    bounds = gdf.bounds.to_numpy()
+    # Calculate mid points based on bounds
+    x_mids, y_mids = _calculate_mid_points(bounds)
+    # Create pairs of x and y midpoints
+    coords = np.array([y_mids, x_mids]).T
+    # Encode coords with Geohash
+    geohash = encode_geohash(coords, string, p)
+
+    return pd.Series(geohash, index=gdf.index, name="geohash")
+
+
+def encode_geohash(coords, string, p):
+    """
+    Calculate geohash based on coordinates for a
+    given precision
+
+    Parameters
+    ----------
+    coords : array_like of shape (n, 2)
+        array of [x, y] pairs
+    string : bool
+        to return string or int Geohash
+    p : int
+        precision of the string Geohash
+    Returns
+    ---------
+    geohash: array containing either int or string
+        geohashes for each mid point
+    """
+
+    quantized_coords = _encode_quantize_points(coords)
+    int_geohash = _encode_into_uint64(quantized_coords)
+
+    if string is True:
```

Comment:
Suggestion: the conditional can be simplified a bit since it involves return statements:

```
if not string:
    return int_geohash

gs_uint8_mat = _encode_base32(int_geohash)
str_geohash = _encode_unicode(gs_uint8_mat, p)
return str_geohash
```